content/en/blog/_posts/2022-10-27-live-and-let-live-with-kluctl-and-ssa.md

```diff
@@ -0,0 +1,174 @@
+available in the [podtato-head](https://github.com/podtato-head/podtato-head) demo project.
+
+## Live and let live
+
+One of the main philosophies that Kluctl follows is ["live and let live"](https://kluctl.io/docs/philosophy/#live-and-let-live),
+meaning that it will try its best to work in conjunction with any other tool or
+controller running outside or inside your clusters. Kluctl will not overwrite
+any fields that it lost ownership of, unless you explicitly tell it to do so.
+
+Achieving this would not have been possible (or at least be several magnitudes
+harder) without the use of Server-side apply. Server-side apply allows Kluctl
+to detect when ownership for a field got lost, for example when another controller
+or operator updates that field to another value. Kluctl can then decide on a 
+field-by-field basis if a force-apply is required and then retry based on these
+decisions.
+
+## The days before Server-side apply
+
+The first versions of Kluctl were based on shelling out to `kubectl` and thus
+implicitly relied on client-side apply. At that time, Server-side apply was
+still alpha and quite buggy. And to be honest, I didn't even know it was a
+thing at that time.
+
+The way client-side apply worked had some serious drawbacks. The most obvious one
+(it was guaranteed that you'd stumble on this by yourself if enough time passed)
+is that it relied on an annotation (kubectl.kubernetes.io/last-applied-configuration)
+being added to the object, bringing in all the limitations and issues with huge
+annotation values. A good example of such issues are
+[CRDs being so large](https://github.com/prometheus-operator/prometheus-operator/issues/4439),
+that they don't fit into the annotation's value anymore.
+
+Another drawback can be seen just by looking at the name (**client**-side apply).
+Being **client** side means that each client has to provide the apply-logic on
+its own, which at that time was only properly implemented inside `kubectl`,
+making it hard to be replicated inside controllers.
+
+This added `kubectl` as a dependency (either as an executable or in the form of
+Go packages) to all controllers that wanted to leverage the apply-logic.
+
+However, even if one managed to get client-side apply running from inside a
+controller, you ended up with a solution that gave no control about how it
+worked internally. As an example, there was no way to individually decide which
+fields to overwrite in case of external changes and which ones to let go. 
+
+## Discovering Server-side apply
+
+I was never happy with the solution described above and then somehow stumbled
+across [server-side apply](/docs/reference/using-api/server-side-apply/),
+which was still in beta at that time. Experimenting with it via
+`kubectl apply --server-side` revealed immediately that the true power of
+server-side apply can not be easily leveraged by shelling out to `kubectl`.
+
+The way server-side apply is implemented in `kubectl` does not allow enough
```

Comment:
Be consistent in all cases when writing `server-side apply`. Choose `Server-side apply` or `server-side apply`

Reply:
Same thing was mentioned by another reviewer as well so I changed it to be `server-side apply` everywhere. Or did I miss something?

Reply:
There are some occurrences of `Server-side apply` above my comment

Reply:
Maybe it's simpler to mention it once and then use the short version of it `SSA`.

Reply:
Oh now I've found the occurences of "Server-side"...strange that I didn't see them before :D I'm going to replace all but the first occurrence with SSA now. I've also verified what is used in other blog posts and SSA is indeed the term mostly used.